q2_metadata/normalization/_norm_check_rules.py

```diff
@@ -0,0 +1,312 @@
+
+from q2_metadata.normalization._norm_errors import (
+    ExpectedError, OntologyError, RemapError,
+    ValidationError, NormalizationError,
+    BlankError, MissingError, FormatError
+)
+
+
+def check_rule(variable: str, rules: dict, rule: str, rule_value):
+    """For each rule in the current variable's rules set,
+    perform checks that the rule values to be used for the rules
+    to apply are properly formatted.
+        - If properly formatted: full the rules dict
+        for the variable's Rules() class instance.
+        - If not properly formatted: raise a specific error and
+        collect the encountered error in the ErrorsCollection()
+        instance for this variable's Rules() class instance.
+
+    Parameters
+    ----------
+    variable : str
+        Name of the metadata variable for which there is rules.
+    rules : dict
+        Rules in the hard-coded data structure that has
+        common default values for all variables and
+        that is checked and updated here.
+    rule : str
+        Current rule name. Could be one of these:
+        - expected: accepted, controlled vocabulary from AGP.
+        - ontology: accepted, controlled vocabulary from a third-party resource.
+        - remap: the replacements to perform ( "A" -> "B" )
+        - validation: the
+        - normalization:
+        - blank:
+        - missing:
+        - format:
+    rule_value : str, list or dict
+        Parsed rule value for the current rule.
+
+    """
+    if rule == 'expected':
+        if check_expected(rule_value):
+            rules['lookups'][rule] = rule_value
+        else:
+            raise ExpectedError(variable, rule_value)
+
+    elif rule == 'ontology':
+        if check_ontology(rule_value):
+            rules['lookups'][rule] = rule_value
+        else:
+            raise OntologyError(variable, rule_value)
+
+    elif rule == 'remap':
+        if check_remap(rule_value):
+            rules['edits'][rule] = rule_value
+        else:
+            raise RemapError(variable, rule_value)
+
+    elif rule == 'validation':
+        if check_validation(rule_value):
+            rules['edits'][rule] = rule_value
+        else:
+            raise ValidationError(variable, rule_value)
+
+    elif rule == 'normalization':
+        if check_normalization(rule_value):
+            rules['edits'][rule] = rule_value
+        else:
+            raise NormalizationError(variable, rule_value)
+
+    elif rule == 'blank':
+        if check_str(rule_value):
+            rules['allowed'][rule] = rule_value
+        else:
+            raise BlankError(variable, rule_value)
+
+    elif rule == 'missing':
+        if check_str(rule_value):
+            rules['allowed'][rule] = rule_value
+        else:
+            raise MissingError(variable, rule_value)
+
+    elif rule == 'format':
+        if check_str(rule_value):
+            rules['format'] = rule_value
+        else:
+            raise FormatError(variable, rule_value)
```

Comment:
Further changes to simplify the lookup `rule_check_map` dictionary - now in the base code that's going top be pushed in an upcoming commit.


```suggestion

    rule_type_map = {
        'expected': 'lookups',
        'ontology': 'lookup',
        'remap': 'edits',
        'validation': 'edits',
        'normalization': 'edits',
        'blank': 'allowed',
        'missing': 'allowed',
        'format': 'format',
    }

    if rule not in rule_type_map:
        raise RuleError(rule, variable, rule_value)

    if rule in ['blank', 'missing']:
        rule_method_key = 'check_blank_missing'
    else:
        rule_method_key = 'check_%s' % rule

    rule_method = globals()[rule_method_key]
    rule_error = globals()['%sError' % rule.capitalize()]
    rule_type = rule_type_map[rule]
    if rule_method(rule_value):
        rules[rule_type][rule] = rule_value
    else:
        raise rule_error(variable, rule_value)
```